braidpool-primitives/src/bead/mod.rs

```diff
@@ -1,14 +1,16 @@
 // Standard Imports
-use std::collections::HashSet;
+use std::cell::RefCell;
+use std::collections::{HashMap, HashSet};
 
 // Bitcoin primitives
 use bitcoin::absolute::Time;
 use bitcoin::transaction::TransactionExt;
 use bitcoin::{BlockHeader, CompactTarget, Transaction};
 
 // Custom Imports
-use crate::utils::BeadHash;
+use crate::braid::Braid;
 use crate::utils::bitcoin::MerklePathProof;
+use crate::utils::{BeadHash, BeadLoadError, Children, Parents};
 
 // Type Aliases
 type TransactionWithMerklePath = (Transaction, MerklePathProof);
```

Comment:
How and when do we plan to use a Merkle Path proof here?

Txs are included in a bead (and there should be a small Merkle tree with the ~2-5 txs so they can be easily committed to).

Merkle proofs are really only used when you need to communicate compactly with another party who doesn't have the same database that we do. As long as we construct our database consistently, we don't need to prove to *ourselves* that a tx was included -- we can just do a db query to verify it, and if we want we can chase the container bead, its merkle tree, etc to verify if we think there's a reason our db is wrong.

I can see ways we might use merkle proofs in the future (settlement/payouts), but I'm not sure how to grok Merkle paths in the basic Braid/Bead structure...